text/0046-404-405-http-error-code.md

```diff
@@ -0,0 +1,121 @@
+| 404 HTTP response status code                       |       x        |                                         |                    |
+| 405 HTTP response status code                       |                |                                         |          x         |
+| 400 HTTP response status code                       |                |                    x                    |                    |
+| Presence of a well formed reponse format            |                |                                         |                    |
+
+The next parts will list the desired behaviors for the release of version 0.21.
+
+| 0.21 Behaviors proposal                             | Invalid url    | Valid url but object not found          | Method not allowed |
+|-----------------------------------------------------|----------------|-----------------------------------------|--------------------|
+| 404 HTTP response status code                       |       x        |                     x                   |                    |
+| 405 HTTP response status code                       |                |                                         |          x         |
+| Presence of a well formed reponse format            |       x        |                     x                   |          x         |
+
+#### 404 Code
+
+The API must return a `404` error code when the resource does not exist or when the url used does not allow to reach a resource.
+
+E.g. Asking for an inexistent ressource.
+
+**GET `/indexes/:index_uid/documents/2`**
+
+In that case, the response body contain a `document_not_found` errorCode.
+
+E.g.
+
+```
+{
+    "message": "Document with id 2 not found",
+    "errorCode": "document_not_found",
+    "errorType": "invalid_request_error",
+    "errorLink": "https://docs.meilisearch.com/errors#document_not_found"
+}
+```
+
+E.g. Using a wrong url.
+
+**GET `/indexess/documents/2`**
+
+In that case, the response body should contain a `unrecognized_request_url` errorCode.
+
+```
+{
+    "message": "Unrecognized request URL (GET: /indexess/documents/2). See https://docs.meilisearch.com/reference/api",
+    "errorCode": "unrecognized_request_url",
+    "errorType": "invalid_request_error",
+    "errorLink": "https://docs.meilisearch.com/errors#unrecognized_request_url"
+}
+```
+
+The message pattern proposal is `Unrecognized request URL ({{HTTP_VERB}}: {{REQUEST URL}}). See https://docs.meilisearch.com/reference/api`
+
+#### 405 Code
+
+The API must return a `405` error code when the resource is reachable but the HTTP verb is not handled by the resource.
+
+E.g. Using a http verb that is not allowed
+
+**DELETE `/indexes/movies/search`**
+
+In that case, the response body should contain a `method_not_allowed` errorCode.
+
+```
+{
+    "message": "Method DELETE is not allowed. See https://docs.meilisearch.com/reference/api/search.html#search",
```

Comment:
Same, the `errorLink` should already redirect the user.

Reply:
If the `errorLink` redirect to the search page (because it was what the user intended) it's ok...but we are loosing the link to the specific error page. If we redirect to the error page instead of the endpoint page, I think the user will lose the error context. That's why I added a contextualized link in the message part.